src/palladio/AnnotationParsing.cpp

```diff
@@ -270,10 +268,21 @@ AttributeAnnotationInfo getAttributeAnnotationInfo(const std::wstring& attribute
 					annotation = currAnnotation;
 					attributeTrait = currAttributeTrait;
 				}
-			}	
+			}
 			else {
-				annotation = currAnnotation;
-				attributeTrait = currAttributeTrait;
+				//keep annotation of range and trait of percent/angle
+				const bool keepAnnotation = (((currAttributeTrait == AttributeTrait::PERCENT) ||
```

Comment:
the handling of percent/angle/range is hard to read and somewhat convoluted... how about just parsing the annotation traits into an enum set and then handling this when creating the spare parms?

Reply:
Now using a different approach with a map from the type enum to a std::variant that contains the annotation information.